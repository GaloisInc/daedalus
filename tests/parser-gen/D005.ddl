-- test grammar Call

def A = { $['a']; $['b'] }

def B = { $['b']; $['c'] }

def Main =
  { Choose
    { x = { A ; $['c'] }            -- abc
    ; y = { $['a']; B; $['d'] } -- abcd
    }
  ; END
  }
