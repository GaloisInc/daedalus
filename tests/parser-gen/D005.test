D005.ddl
--dump-gen
