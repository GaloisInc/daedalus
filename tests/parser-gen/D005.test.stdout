Stack:
    *
    Main callsite __START__
Start: Main (12,5)-(12,7)
SlkCfg{ q:72, ctrl:[?, *], sem:[?, ], inp:--- }
DTrans [
  ( [(9),]
  , EndInput
  , Resolution (NotAmbiguous)
  ),
]

Stack:
    *
Start: Main (7,1)-(13,3)
SlkCfg{ q:1, ctrl:[*], sem:[*], inp:--- }
DTrans [
  ( [(19),(15),]
  , HeadInput ['a']
  , DTrans [
      ( [(9),(9),]
      , HeadInput ['b']
      , DTrans [
          ( [(9),(10),]
          , HeadInput ['c']
          , DTrans [
              ( [(11),]
              , HeadInput ['d']
              , Resolution (NotAmbiguous)
              ),
              ( [(9),]
              , EndInput
              , Resolution (NotAmbiguous)
              ),
            ]
          ),
        ]
      ),
    ]
  ),
]

Stack:
    *
Start: __START__
SlkCfg{ q:0, ctrl:[*], sem:[?, *], inp:--- }
DTrans [
  ( [(0),]
  , AcceptingPath
  , NotAmbiguous
  ),
]

**********************
***** Strict LLA *****
**********************
("Lookahead-0",1)
("Lookahead-1",1)
("Lookahead-4",1)

Total nb DFAs: 3

SUCCESS: Fully determinized format
